### PR TITLE
Implement new MigrationBuilder API

### LINKED ISSUE #1654: Down() method in migration is generated with wrong order.
public partial class Create_Languages_Table : Migration
    {
        public override void Up(MigrationBuilder migrationBuilder)
        {
            migrationBuilder.CreateTable(....)
                .PrimaryKey(x => x.LanguageId, name: "PK_Language");

            migrationBuilder.AddColumn("AspNetUsers", "LanguageId", x => x.Column("int", nullable: true));

            migrationBuilder.AddForeignKey("AspNetUsers", "LanguageId", "Language", principalColumns: "LanguageId", name: "FK_AspNetUsers_Language_LanguageId");
        }

        public override void Down(MigrationBuilder migrationBuilder)
        {
            migrationBuilder.DropColumn("AspNetUsers", "LanguageId");
            migrationBuilder.DropForeignKey("AspNetUsers", "FK_AspNetUsers_Language_LanguageId");
            migrationBuilder.DropTable("Language");
        }
    }

DropForeignKey must be before DropColumn. 
And principalColumns: "LanguageId" must be principalColumn: "LanguageId".

### LINKED ISSUE #1777: Metadata: Add cycle option to Sequence
The cycle option (`CYCLE` or `NO CYCLE`) is part of the SQL standard (ISO 9075) for sequence generators, but not part of our [Sequence](https://github.com/aspnet/EntityFramework/blob/dev/src/EntityFramework.Relational/Metadata/ISequence.cs) metadata. We should consider adding it.

### LINKED ISSUE #1823: Migrations: Make MigrationBuilder API more extensible
Similar to the changes we made in the Code First Fluent API, we need to make individual items addressable from provider-specific extension methods.

### LINKED ISSUE #1912: Migrations: Use EXEC when calling sp_rename
Otherwise it will only work if it's the first statement in the batch.

### PR DESCRIPTION
**Pull request by bricelam**
Resolves #1823, resolves #1777, fixes #1654 & fixes #1912

### CODE REVIEW

**Comment by ajcvickers** [comment]
:sheep: :it: Really like the new API.

File: src/EntityFramework.SqlServer/Extensions/SqlServerMigrationBuilderExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using JetBrains.Annotations;
+using Microsoft.Data.Entity.Relational.Migrations.Operations;
+using Microsoft.Data.Entity.SqlServer.Metadata;
+
+namespace Microsoft.Data.Entity.Relational.Migrations.Builders
+{
+    public static class SqlServerMigrationBuilderExtensions
```
Comment (bricelam) [inline src/EntityFramework.SqlServer/Extensions/SqlServerMigrationBuilderExtensions.cs]: I think this class should be removed for now.

**Comment by bricelam** [comment]
Updated based on design meeting feedback. Can someone review a6d4e70b3c658648d3f49a1f546db727e403378f?

**Comment by ajcvickers** [comment]
:shipit:

**Pull request merged**